src/SoftRobots/component/constraint/model/CableModel.inl

```diff
@@ -261,6 +322,242 @@ void CableModel<DataTypes>::checkIndicesRegardingState()
+            }
+            
+            // Compute barycentric coordinates for visualization purposes
+            Real alpha; Real beta;
+            computeBarycentric(closestTriangle, closestProjectionOnTriangle, alpha, beta);
+            m_alpha[i] = alpha;
+            m_beta[i] = beta;
+            m_closestTriangle[i] = closestTriangle;
+
+
+        } else {
+            distances[indices[i]] = 0.0;
+            queue.insert(DistanceToPoint(0.0,indices[i]));
+        }
+
+        // Dijkstra
+        while(!queue.empty() )
+        {
+            DistanceToPoint top = *queue.begin();
+            queue.erase(queue.begin());
+            BaseMeshTopology::PointID v = top.second; 
+            const vector<BaseMeshTopology::PointID> neighboors = m_topology->getVerticesAroundVertex(v);
+            for (unsigned int j=0 ; j<neighboors.size(); ++j)
+            {
+                sofa::core::topology::BaseMeshTopology::PointID vn = neighboors[j];
+
+                Coord pv; 
+                getPositionFromTopology(pv, v);
+                Coord pvn;
+                getPositionFromTopology(pvn, vn);
+                Real d = 0.0;
+                if (id_method == 2)
+                    d = distances[v] + (pv - pvn).norm();
+                else if (id_method == 1)
+                    d = (pvn - cablePositions[indices[i]]).norm();
+
+                if((distances[vn]) > d)
+                {
+                    qit=queue.find(DistanceToPoint(distances[vn],vn));
+                    if(qit != queue.end()) queue.erase(qit);
+                    distances[vn] = d;
+                    queue.insert(DistanceToPoint(d,vn) );
+                }
+            }
+        }
+
+        // Distribute forces
+        m_totalRatio[i] = 0.0;
+        for(unsigned int p=0; p<distances.size(); p++)
+        {
+            if(distances[p] < maxDistance[i])
+            {
+                auto ratio = rabs(distances[p] - maxDistance[i]);
+
+                m_totalRatio[i] += ratio;
+                m_ratios[i].push_back(ratio);
+                m_areaIndices[i].push_back(p);
+            } 
+        }
+
+        // Normalize force ratios
+        for(unsigned int j=0; j<m_ratios[i].size(); j++)
+            m_ratios[i][j] = m_ratios[i][j] / m_totalRatio[i];
+    }
+}
+
+
+template<class DataTypes>
+unsigned int CableModel<DataTypes>::computeClosestIndice(Coord position)
+{
+    Coord p0;
+    getPositionFromTopology(p0, 0);
+    auto closest_distance = (position - p0).norm();
+    unsigned int closest_vertice = 0;
+    for(unsigned int j=1; j<m_topology->getNbPoints(); j++)
+    {
+        Coord pj;
+        getPositionFromTopology(pj, j);
+        const auto distance = (position - pj).norm();
+        if(distance < closest_distance) 
+        {
+            closest_distance = distance;
+            closest_vertice = j;
+        }
+    } 
+   return closest_vertice;
+}
+
+
+template<class DataTypes>
```

Comment:
In the two following functions, you suppose that DataTypes is Vec3. But it is not clear. DataTypes is a template parameter , so it could be anything. You should specialize those functions in the cpp file